### ISSUE TITLE
GWEN is not able to handle "if" word in user defined DSLs

### ISSUE THREAD

**Issue by sse2ebdd** [gwen-interpreter/gwen-web#93]
Steps to reproduce -

1. Create feature file with below content and run it
```
Feature: Test

@StepDef
Scenario: I check if the dashboard is displaying "<role>" role
And role name is "$<role>"

Scenario: Test
And I check if the dashboard is displaying "Admin" role
```
2. GWEN fails with error --> **Unbound reference: the dashboard is displaying "Admin" role**
3. This used to work with earlier versions of GWEN such as 2.53.6

"If" word can be used in user defined DSL as a natural language, therefore GWEN should still support it.

Env -
GWEN version - 3.0.10
Chrome driver - Latest (96)

**Comment by bjuric** [comment]
Fixed in v[3.0.11](https://github.com/gwen-interpreter/gwen-web/releases/tag/v3.0.11). Thanks

**Comment by sse2ebdd** [comment]
@bjuric 
I got similar issues with 'while' and 'in' words.
Feature -
```
Feature: Test

@StepDef
Scenario: I validate error while creating new dashboard
And something is "value"

@StepDef
Scenario: I validate hyperlink is available for each of an entity ID in Timeline summary
And something is "value"

Scenario: Test 'while' word 
And I validate error while creating new dashboard

Scenario: Test 'in' word
And I validate hyperlink is available for each of an entity ID in Timeline summary
```

Error for 'while' - Unbound reference in feature scope: creating new dashboard/javascript [at BDD/features/test.feature:14]

Error for 'in' - Undefined selector for: Timeline summary

**Comment by sse2ebdd** [comment]
One more issue related to "if" word.

`And value is "This will fail if it contains if" -->  Unbound reference: it contains if`

**Comment by bjuric** [comment]
Please try this snapshot to see if it fixes all of the above:
- [gwen-web-3.0.11-1-gb21abaa-SNAPSHOT.zip](https://oss.sonatype.org/content/repositories/snapshots/org/gweninterpreter/gwen-web/3.0.11-1-gb21abaa-SNAPSHOT/gwen-web-3.0.11-1-gb21abaa-SNAPSHOT.zip)

**Comment by sse2ebdd** [comment]
Found another issue on provided snapshot version.
Run below feature -
```
Feature: Test

@StepDef
Scenario: some function "<param1>"
And attr is "$<param1>"

Scenario: Scenario name
And strings is "some string1, some string2, some string3"
And some function "${str}" for each str in strings delimited by ","
```

Param 1 should be resolved as --> some string1
But GWEN resolves it as --> ${str}" for each str in strings delimited by ",

GWEN version - gwen-web-3.0.11-1-gb21abaa-SNAPSHOT.zip

**Comment by bjuric** [comment]
I tested above using the snapshot and each string resolved OK. Here is a snapshot of the report I got.

![image](https://user-images.githubusercontent.com/1369994/145655700-0e66ad18-84f5-4e07-b049-10472d4f041e.png)

**Comment by bjuric** [comment]
However, this part doesn't seem right, which you're referring to. I'll look into it. 

![image](https://user-images.githubusercontent.com/1369994/145656545-a0d872d0-c78c-49e1-917b-e2ac87729e79.png)

**Comment by bjuric** [comment]
Please retest with this snapshot:
- [gwen-web-3.0.11-2-gb21abaa-SNAPSHOT.zip](https://oss.sonatype.org/content/repositories/snapshots/org/gweninterpreter/gwen-web/3.0.11-2-gb21abaa-SNAPSHOT/gwen-web-3.0.11-2-gb21abaa-SNAPSHOT.zip)

**Comment by bjuric** [comment]
Fixed in v[3.0.12](https://github.com/gwen-interpreter/gwen-web/releases/tag/v3.0.12)

**Comment by sse2ebdd** [comment]
@bjuric 
In latest GWEN 3.1.1 above issues got fixed.
However, I got similar new issue.
When we override GWEN default DSL and use it along with 'if' condition GWEN fails.

Run below feature
```
Feature: Test

#we have overridden GWEN default DSL
@StepDef
Scenario: I enter "<text>" in <elem>
And I clear $<elem>
And I enter "$<text>" in $<elem>

Scenario: Test
And add roles textfield can be located by xpath "//input"
And role not added is defined by javascript "true"
And I enter "RoleName" in add roles textfield if role not added
```

It gives error --> Undefined selector for: add roles textfield if role not added

**Comment by bjuric** [comment]
Fixed in v[3.1.2](https://github.com/gwen-interpreter/gwen-web/releases/tag/v3.1.2)

**Comment by sse2ebdd** [comment]
Verified this issue on GWEN v3.1.2.
It's working as expected, hence closing this issue.
